test/rpc_test.py

```diff
@@ -914,6 +922,50 @@ def test_rref_forward_chain(self):
         ret = ret_rref
         self.assertEqual(ret, torch.add(torch.ones(n, n), 1))
 
+    @dist_init
+    def test_local_rref_no_fork(self):
+        local_rref = rpc.RRef(35)
```

Comment:
This API is a little weird. This RRef has nothing to do with rpc, yet we're creating it with `rpc.RRef()`. I think RRef should be a separate package since it is independent from RPC (as we can see here). If we want to allow creating local RRefs like this, RRef shouldn't be tightly coupled with RPC.

Reply:
There's 2 things in your comment:

1. would you prefer to have simply torch.distributed.RRef ?  I can do that.

2. Currently init_rpc() needs to be called before we can call RRef() constructor even locally since only OnwerRRef constructor is private and only accessible by RRefContext. I can change that too if you think is appropriate? In that case, RRefs would become known to RRefContext only at fork time.

Reply:
1. Yes, I think that makes sense. 
2. I think this is why we probably need a higher level initialization like `init_model_parallel` (I'm open to a better name) that initializes rpc, rref and autograd. The user has to call this API even if they want to only use rpc/rref/dist_autograd individually.

Reply:
I thought a bit about this -- I don't think that it makes sense to initialize RRef without initializing RPC, actually. RRef semantic requires RPC concepts such as owner() . owner() only makes sense if an RPCAgent is present.

RPC without RRef doesn't make sense either -- since RRef is a native type supported as both input and output of any RPC calls, particularly Remote.

As for 1., I'll change it.

Reply:
Please don't do 1. The top level `torch.distributed` package concerns only collectives. As you say, RPC without RRef and RRef without RPC are both useless. If that ever changes, we could consider promoting RRefs to their own package, but not to the top level `torch.distributed` module.

Reply:
Update: I see they're part of `torch.distributed` today, but that's because that's where they were defined from the very beginning. I have #27529 out (pending type checking fixes) to update that and move the pybind11 wrappers to the appropriate modules. We can discuss moving them around, but let's use a new issue for that.

Reply:
Ok i'm just going to reuse the existing distributed.RRef for now. Will leave to @pietern to move it afterwards.